File: java/Easy/_392_IsSubsequence.java
package Easy;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.testng.annotations.Test;

public class _392_IsSubsequence {
    public boolean isSubsequence(String s, String t) {
        int cursorS = 0;
        if("".equals(s)) return true;
        for(int i = 0; i<t.length(); i++){
            if(s.charAt(cursorS) == t.charAt(i)){
                cursorS++;
            }
            if(cursorS>=s.length()) return true;
        }
        return false;
    }

    @Test
    public void test(){
        assertTrue(isSubsequence("abc", "ahbgdc"));
        assertFalse(isSubsequence("axc", "ahbgdc"));
    }
}

File: Python3/_392_IsSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == "" :
            return True
        csS = 0
        for c in t:
            if c is s[csS]:
                csS+=1
            if csS >= len(s):
                return True
        return False

test = Solution()
print(test.isSubsequence("abc","adddbdsdcdd"))